libsolutil/CDCL.cpp

```diff
@@ -121,7 +136,7 @@ optional<Clause> CDCL::propagate()
 			if (clause.front() != falseLiteral)
 				swap(clause[0], clause[1]);
 			solAssert(clause.front() == falseLiteral);
-			if (clause.size() >= 2 && isAssignedTrue(clause[1]))
+			if (value(clause[1]) == TriState::t_true())
```

Comment:
```suggestion
			if (value(clause[1]) == TriState{true})
```
(shorter and does not need the constexpr function)

Reply:
Yep, I fixed it. Actually, I can remove the constexpr in retrospect. This is cleaner.

Reply:
Thanks for sticking to this. You were right, it's cleaner this way :)